docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md

```diff
@@ -14,305 +14,194 @@ weight: 700
+| `team_ids_attribute_path`    | No       | [JMESPath](http://jmespath.org/examples.html) expression to use for Grafana team ID lookup within the results returned by `teams_url` endpoint. Used together with `team_ids` and `teams_url`. |                 |
+| `teams_url`                  | No       | URL used to query for team IDs. If not set, defaults to `/teams`. Used together with `team_ids` and `team_ids_attribute_path`.                                                                 |                 |
+| `tls_skip_verify_insecure`   | No       | If set to `true`, client will not verify the server's certificate chain and host name. Should only be set to `true` for testing purposes.                                                      | `false`         |
+| `tls_client_cert`            | No       | Path of the certificate.                                                                                                                                                                       |                 |
+| `tls_client_key`             | No       | Path to the key.                                                                                                                                                                               |                 |
+| `tls_client_ca`              | No       | Path to the trusted certificate authority list.                                                                                                                                                |                 |
+| `use_pkce`                   | No       | Set to `true` to use "proof key for code exchange" (PKCE).                                                                                                                                     | `false`         |
+
+### Auth style
+
+`auth_style` controls which [OAuth2 AuthStyle](https://pkg.go.dev/golang.org/x/oauth2#AuthStyle) is used when token is requested from OAuth2 provider. It determines how `client_id` and `client_secret` are sent to Oauth2 provider.
+Available values are `AutoDetect`, `InParams` and `InHeader`. By default, `AutoDetect` is used.
 
-```bash
-[auth.generic_oauth]
-name = OAuth
-icon = signin
-enabled = true
-allow_sign_up = true
-auto_login = false
-client_id = YOUR_APP_CLIENT_ID
-client_secret = YOUR_APP_CLIENT_SECRET
-scopes =
-empty_scopes = false
-auth_url =
-token_url =
-api_url =
-allowed_domains = mycompany.com mycompany.org
-allowed_groups = ["Admins", "Software Engineers"]
-tls_skip_verify_insecure = false
-tls_client_cert =
-tls_client_key =
-tls_client_ca =
-use_pkce = true
-auth_style =
-```
+### Email address
 
-Set `api_url` to the resource that returns [OpenID UserInfo](https://connect2id.com/products/server/docs/api/userinfo) compatible information.
+Grafana determines a user's email address by following the steps below until it finds and e-mail address:
```

Comment:
This section contains a lot of technical detail that seems not relevant to what the user needs to do. Why is this information important to the user?